app/lib/pre_assembly/file_manifest.rb

```diff
@@ -0,0 +1,133 @@
+# cm=PreAssembly::FileManifest.new(bundle_dir: File.join(Rails.root,'spec/test_data/media_audio_test'), csv_filename: 'file_manifest.csv', verbose: true)
+# puts cm.generate_cm('sn000dd0000', :media)
+
+# or in the context of a batch object:
+# cm=PreAssembly::FileManifest.new(csv_filename: @content_md_creation[:file_manifest],bundle_dir: @bundle_dir, verbose: false)
+# puts cm.generate_cm('oo000oo0001', :file)
+
+module PreAssembly
+  class FileManifest
+    attr_reader :manifest, :rows, :csv_filename, :bundle_dir
+
+    # the valid roles a file can have, if you specify a "role" column and the value is not one of these, it will be ignored
+    VALID_ROLES = %w[transcription annotations derivative master].freeze
+
+    def initialize(params)
+      @bundle_dir = params[:bundle_dir]
+      csv_file = params[:csv_filename] || 'file_manifest.csv'
+      @csv_filename = File.join(@bundle_dir, csv_file)
+      @manifest = {}
+      # read CSV
+      load_manifest # this will cache the entire manifest in @rows and @manifest
+    end
+
+    def load_manifest
+      # load file into @rows and then build up @manifest
+      @rows = CsvImporter.parse_to_hash(@csv_filename)
+      @rows.each do |row|
+        druid = row[:druid]
+        file_extension = File.extname(row[:filename])
+        resource_type = row[:resource_type]
+
+        # set the role for the file (if a valid role value, otherwise it will be left off)
+        role = row[:role] if VALID_ROLES.include?(row[:role])
+
+        # set the thumb attribute for this resource - if it is set in the manifest to true, yes or thumb (set to false if no value or column is missing)
+        thumb = row[:thumb] && %w[true yes thumb].include?(row[:thumb].downcase) ? true : false
+
+        # set the publish/preserve/shelve
+        publish  = row[:publish]
+        shelve   = row[:shelve]
+        preserve = row[:preserve]
+
+        manifest[druid] ||= { files: [] }
+        files_hash = { file_extention: file_extension, filename: row[:filename], label: row[:label], sequence: row[:sequence] }
+        manifest[druid][:files] << files_hash.merge(role: role, thumb: thumb, publish: publish, shelve: shelve, preserve: preserve, resource_type: resource_type)
+      end
+    end
+
+    # actually generate content metadata for a specific druid in the manifest
+    # @return [String] XML
+    def generate_cm(druid, content_md_creation_style)
+      return '' unless manifest[druid]
+      current_directory = Dir.pwd
+      files = manifest[druid][:files]
+      current_seq = ''
+      resources = {}
+
+      # group the files into resources based on the sequence number defined in the manifest
+      #  a new sequence number triggers a new resource
+      files.each do |file|
+        seq = file[:sequence]
+        label = file[:label] || ''
+        resource_type = file[:resource_type]
+        if !seq.nil? && seq != '' && seq != current_seq # this is a new resource if we have a non-blank different sequence number
```

Comment:
I wonder if replacing `!seq.nil? && seq != ''` with `seq.present?` would be appropriate?

Reply:
Sounds good.  Note that ~99% of the code in this file didn't change at all -- the class was just renamed with a few small edits, but git decided to delete the old one and treat this as a brand new file.

Reply:
The file used to be `app/lib/pre_assembly/media.rb`

Reply:
Yeah, I see that it's a lot of deletion. :-)